server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -109,35 +105,32 @@ public boolean equals(Object o) {
                 return false;
             }
             Response response = (Response) o;
-            return clusterName.equals(response.clusterName) && status == response.status && components.equals(response.components);
+            return clusterName.equals(response.clusterName)
+                && status == response.status
+                && indicatorResults.equals(response.indicatorResults);
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(clusterName, status, components);
+            return Objects.hash(clusterName, status, indicatorResults);
         }
 
         @Override
         public String toString() {
-            return "Response{clusterName=" + clusterName + ", status=" + status + ", components=" + components + '}';
+            return "Response{clusterName=" + clusterName + ", status=" + status + ", indicatorResults=" + indicatorResults + '}';
         }
     }
 
     public static class Request extends ActionRequest {
-        private final String componentName;
         private final String indicatorName;
         private final boolean explain;
 
         public Request(boolean explain) {
-            // We never compute details if no component name is given because of the runtime cost:
```

Comment:
Should we preserve this comment? If I understand correctly the same applies for the indicators, right?

Reply:
Oh good point. I'll restore it.